spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubTemplate.java

```diff
@@ -40,39 +41,68 @@
 import org.springframework.beans.factory.InitializingBean;
 import org.springframework.cloud.gcp.pubsub.support.PublisherFactory;
 import org.springframework.cloud.gcp.pubsub.support.SubscriberFactory;
+import org.springframework.cloud.gcp.pubsub.support.converter.JacksonMessageConverter;
+import org.springframework.cloud.gcp.pubsub.support.converter.PubSubMessageConverter;
 import org.springframework.util.Assert;
 import org.springframework.util.concurrent.ListenableFuture;
 import org.springframework.util.concurrent.SettableListenableFuture;
 
 /**
+ * // @formatter:off
  * Default implementation of {@link PubSubOperations}.
  *
- * <p>The main Google Cloud Pub/Sub integration component for publishing to topics and consuming messages from
- * subscriptions asynchronously or by pulling.
- *
+ * <p>The main Google Cloud Pub/Sub integration component for publishing to topics and
+ * consuming messages from subscriptions asynchronously or by pulling.
+ * // @formatter:on
  * @author Vinicius Carvalho
  * @author João André Martins
  * @author Mike Eltsufin
+ * @author Chengyuan Zhao
  */
 public class PubSubTemplate implements PubSubOperations, InitializingBean {
 
 	private static final Log LOGGER = LogFactory.getLog(PubSubTemplate.class);
 
+	private final PubSubMessageConverter messageConverter;
+
 	private final PublisherFactory publisherFactory;
 
 	private final SubscriberFactory subscriberFactory;
 
 	/**
-	 * Default {@link PubSubTemplate} constructor.
+	 * {@link PubSubTemplate} constructor.
 	 *
 	 * @param publisherFactory the {@link com.google.cloud.pubsub.v1.Publisher} factory to
-	 *                         publish to topics
-	 * @param subscriberFactory the {@link com.google.cloud.pubsub.v1.Subscriber} factory to
-	 *                          subscribe to subscriptions
+	 * publish to topics
+	 * @param subscriberFactory the {@link com.google.cloud.pubsub.v1.Subscriber} factory
+	 * to subscribe to subscriptions
+	 * @param messageConverter a converter for converting Java objects to and from message
+	 * payloads.
 	 */
-	public PubSubTemplate(PublisherFactory publisherFactory, SubscriberFactory subscriberFactory) {
+	public PubSubTemplate(PublisherFactory publisherFactory,
+			SubscriberFactory subscriberFactory,
+			PubSubMessageConverter messageConverter) {
 		this.publisherFactory = publisherFactory;
 		this.subscriberFactory = subscriberFactory;
+		this.messageConverter = messageConverter;
+	}
+
+	public PubSubMessageConverter getMessageConverter() {
+		return this.messageConverter;
+	}
+
+	/**
+	 * {@link PubSubTemplate} constructor that uses a {@link JacksonMessageConverter} to
+	 * serialize and deserialize payloads.
+	 *
```

Comment:
No blank lines in the method JavaDocs: https://github.com/spring-projects/spring-framework/wiki/Code-Style#javadoc-formatting

Reply:
done